### PR TITLE
Refactor `Provider#initialize` to parametrize configuration

### PR DESCRIPTION
**Pull request by antaz**
## Changes
- Allow `host`, `port` and `ssl` to be passed to provider initializer

### CODE REVIEW

File: lib/llm/providers/anthropic.rb [hunk 1]
```diff
@@ -13,8 +13,8 @@ class Anthropic < Provider
 
     ##
     # @param secret (see LLM::Provider#initialize)
-    def initialize(secret)
-      super(secret, HOST)
+    def initialize(secret, host: HOST)
```
Comment (0x1eef) [inline lib/llm/providers/anthropic.rb:16]: I think we want something like this - right ?

```ruby
def initialize(secret, **options)
  super(secret, **{host: HOST}.merge(options))
end
```

File: lib/llm/provider.rb [hunk 2]
```diff
@@ -14,10 +14,10 @@ class Provider
     #  The host address of the LLM provider
     # @param [Integer] port
     #  The port number
-    def initialize(secret, host, port = 443)
+    def initialize(secret, **options)
```
Comment (0x1eef) [inline lib/llm/provider.rb:17]: We could be more strict about what's accepted ? `host`, `port`, `ssl`.

File: lib/llm/providers/anthropic.rb (hunk 1 above)
Comment (antaz) [reply lib/llm/providers/anthropic.rb:16]: OH! true, what a blindspot

File: lib/llm/provider.rb (hunk 2 above)
Comment (antaz) [reply lib/llm/provider.rb:17]: @0x1eef How?

Comment (0x1eef) [reply lib/llm/provider.rb:17]: `LLM::Provider` could define `initialize` similar to this:
```ruby
def initialize(secret, host:, port: 443, ssl: true)
end
```
Then the subclasses of `LLM::Provider` could be similar to what you already have:
```ruby
def initialize(secret, **)
  # ...
end
```

And if you pass something like `blah`, that should be an error. What do you think?

Comment (antaz) [reply lib/llm/provider.rb:17]: That’s what I thought initially, but i also then thought the options approach was fine

Comment (0x1eef) [reply lib/llm/provider.rb:17]: It works - but then it's more like a Hash than keyword arguments. We can specify exactly which `keys` we accept, and if we're given some kind of rubbish - error out.

File: lib/llm/providers/gemini.rb [hunk 3]
```diff
@@ -13,8 +13,8 @@ class Gemini < Provider
 
     ##
     # @param secret (see LLM::Provider#initialize)
-    def initialize(secret)
-      super(secret, HOST)
+    def initialize(secret, options = {})
+      super(secret, host: HOST, **options)
```
Comment (0x1eef) [inline lib/llm/providers/gemini.rb:17]: Why is this provider different to the others ?

Comment (antaz) [reply lib/llm/providers/gemini.rb]: Fixed

**Review by 0x1eef — APPROVED** [review]
Looks good!

**Pull request merged**